src/python/pants/backend/python/tasks/pex_build_util.py

```diff
@@ -50,6 +50,55 @@ def is_python_binary(tgt):
   return isinstance(tgt, PythonBinary)
 
 
+def tgt_closure_has_native_sources(tgts):
+    """Determine if any target in the current target closure has native (c or cpp) sources."""
+    return any(tgt.has_native_sources for tgt in tgts)
+
+
+def tgt_closure_platforms(tgts):
+  """
+  Aggregates a dict that maps a platform string to a list of targets that specify the platform.
+  If no targets have platforms arguments, return a dict containing platforms inherited from
+  the PythonSetup object.
+
+  :param tgts: a list of :class:`Target` objects.
+  :returns: a dict mapping a platform string to a list of targets that specify the platform.
+  """
+  tgts_by_platforms = {}
+  for tgt in tgts:
+    if tgt.platforms:
+      for platform in tgt.platforms:
+        if platform in tgts_by_platforms:
+          tgts_by_platforms[platform].append(tgt)
+        else:
+          tgts_by_platforms[platform] = [tgt]
+  # If no targets specify platforms, inherit the default platforms.
+  if not tgts_by_platforms:
+    for platform in PythonSetup.global_instance().platforms:
+      tgts_by_platforms[platform] = ['(No target) Platform inherited from either the '
+                                     '--platforms option or a pants.ini file.']
+  return tgts_by_platforms
+
+
+def build_for_current_platform_only_check(context_tgts_func):
+  """
+  Performs a check of whether the current target closure has native sources and if so, ensures that
+  Pants is only targeting the current platform.
+
+  :context_tgts_func function: The target filtering function of the current task context.
```

Comment:
This parameter can just be a list of targets, and then you can call `filter`. There is nothing magical about `self.context.targets()`: if you pass it a predicate, it calls `filter` for you... if you don't, you get everything.